### PR TITLE
Adding cleanup for share backup test and impl of polling for delete u…

### PR DESCRIPTION
**Pull request by sn-px**
…ser validation

**What this PR does / why we need it**:
Adding cleanup for share backup test and impl of polling for delete user

http://aetos.pwx.purestorage.com/resultSet/testSetID/92928

latest log : https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test/151/

### CODE REVIEW

**Comment by mkoppal-px** [comment]
Please fix the compilation failure

File: tests/backup/backup_test.go [hunk 1]
```diff
@@ -6409,19 +6411,21 @@ var _ = Describe("{DeleteUsersRole}", func() {
...
+						if roleObj.Name == string(role) {
+							return "", true, fmt.Errorf("validation of deleted role [%s] from user [%s]", role, user)
+						} else {
+							return "", false, nil
+						}
 					}
+					return "", false, nil
 				}
-				dash.VerifyFatal(result, false, fmt.Sprintf("validation of deleted role [%s] from user [%s]", role, user))
+				_, err := task.DoRetryWithTimeout(roleDeletionSuccess, 3*time.Minute, 10*time.Second)
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:6427]: Reduce the timeout duration

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -6409,19 +6411,21 @@ var _ = Describe("{DeleteUsersRole}", func() {
...
 		Step("Validate if the roles are deleted from the users ", func() {
-			result := false
 			for user, role := range userRoleMapping {
-				roles, _ := backup.GetRolesForUser(user)
-				for _, roleObj := range roles {
-					if roleObj.Name == string(role) {
-						result = true
-						break
+				roleDeletionSuccess := func() (interface{}, bool, error) {
+					roles, _ := backup.GetRolesForUser(user)
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:6417]: Handle the error here

File: tests/backup/backup_test.go [hunk 3]
```diff
@@ -6850,8 +6851,56 @@ var _ = Describe("{SwapShareBackup}", func() {
...
+				}
+			}
+
+			backupLocationDeletionSuccess := func() (interface{}, bool, error) {
+				allBackupLocations, err := getAllBackupLocations(ctx)
+				dash.VerifySafely(err, nil, "Verifying fetching of all backup locations")
+				for _, backupLocationName := range allBackupLocations {
+					if userBackupLocationMapping[userName] == backupLocationName {
+						return "", true, fmt.Errorf("found %s backup locations", backupLocationName)
+					} else {
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:6894]: can you please check this logic. i  think this might fail when there are multiple backup location for one user

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/backup/backup_test.go (hunk 3 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: resolved

File: tests/backup/backup_test.go (hunk 2 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: resolved

File: tests/backup/backup_test.go (hunk 1 above)
Comment (sn-px) [reply tests/backup/backup_test.go]: removed timeout

File: drivers/backup/auth.go [hunk 4]
```diff
@@ -566,6 +567,7 @@ func DeleteRoleFromUser(userName string, role PxBackupRole, description string)
 	fn := "DeleteRoleFromUser"
 	// First fetch the user ID of the user
 	clientID, err := FetchIDOfUser(userName)
+	log.Infof("Response from FetchIDOfUser %s", string(clientID))
```
Comment (nikolaypopov) [inline drivers/backup/auth.go:570]: This function `func FetchIDOfUser(userName string) (string, error)` already return `clidID` as a `string`, don't need to `string(clientID)`

File: tests/backup/backup_test.go [hunk 5]
```diff
@@ -6409,17 +6411,17 @@ var _ = Describe("{DeleteUsersRole}", func() {
...
 			for user, role := range userRoleMapping {
-				roles, _ := backup.GetRolesForUser(user)
+				roles, err := backup.GetRolesForUser(user)
+				log.FailOnError(err, "Failed to get roles for user - %s", user)
 				for _, roleObj := range roles {
 					if roleObj.Name == string(role) {
 						result = true
 						break
 					}
+					log.Info(fmt.Sprintf("Role name from user [%s] added role [%s] user [%s]", roleObj.Name, string(role), user))
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:6424]: This log message is not clear, needs to be re-worded

```
Role name from user [px-backup-app.user] added role [px-backup-infra.admin] user [testautouser1675943288]
```

Comment (sn-px) [reply tests/backup/backup_test.go]: resolved

File: drivers/backup/auth.go (hunk 4 above)
Comment (sn-px) [reply drivers/backup/auth.go]: resolved

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

File: tests/backup/backup_test.go [hunk 6]
```diff
@@ -6850,8 +6850,54 @@ var _ = Describe("{SwapShareBackup}", func() {
...
 			dash.VerifySafely(err, nil, fmt.Sprintf("Verify destroying app %s, Err: %v", taskName, err))
 		}
-		var wg sync.WaitGroup
 
+		log.InfoD("Deleting all restores")
+		for _, userName := range users {
+			ctx, err := backup.GetNonAdminCtx(userName, "Password1")
+			log.FailOnError(err, "Fetching nonAdminctx")
+			allRestores, err := GetAllRestoresNonAdminCtx(ctx)
+			log.FailOnError(err, "Fetching all restors for nonAdminctx")
```
Comment (mkoppal-px) [inline tests/backup/backup_test.go:6859]: Spell check

Comment (sn-px) [reply tests/backup/backup_test.go]: corrected

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**